test/unittests/commands/xmlupload/test_check_consistency_with_ontology_low_level.py

```diff
@@ -293,3 +315,312 @@ def test_get_all_classes_and_properties_from_data() -> None:
+    def test_utf_simple_correct(self) -> None:
+        test_val = TextValueData("id", ":prop", "utf8")
+        test_lookup = PropertyTextValueEncodingTypes(set(), {":prop"})
+        assert _check_correctness_one_prop(test_val, test_lookup) is True
+
+    def test_utf_simple_wrong(self) -> None:
+        test_val = TextValueData("id", ":prop", "utf8")
+        test_lookup = PropertyTextValueEncodingTypes({":prop"}, set())
+        assert _check_correctness_one_prop(test_val, test_lookup) is False
+
+    def test_xml_correct(self) -> None:
+        test_val = TextValueData("id", ":prop", "xml")
+        test_lookup = PropertyTextValueEncodingTypes({":prop"}, set())
+        assert _check_correctness_one_prop(test_val, test_lookup) is True
+
+    def test_xml_wrong(self) -> None:
+        test_val = TextValueData("id", ":prop", "xml")
+        test_lookup = PropertyTextValueEncodingTypes(set(), set())
+        assert _check_correctness_one_prop(test_val, test_lookup) is False
+
+
+def test_analyse_all_text_value_encodings_are_correct_all_good() -> None:
+    test_ele = etree.fromstring(
+        """<knora>
+                <resource label="First Testthing"
+                      restype=":TestThing"
+                      id="test_thing_1"
+                      permissions="res-default">
+                    <uri-prop name=":hasUri">
+                        <uri permissions="prop-default">https://dasch.swiss</uri>
+                    </uri-prop>
+                    <text-prop name=":hasRichtext">
+                        <text encoding="xml">Text</text>
+                    </text-prop>
+                </resource>
+                <resource label="resB" restype=":TestThing2" id="resB" permissions="res-default">
+                    <text-prop name=":hasSimpleText">
+                        <text encoding="utf8">Text</text>
+                    </text-prop>
+                </resource>
+                <resource label="resC" restype=":TestThing2" id="resC" permissions="res-default">
+                    <resptr-prop name=":hasResource2">
+                        <resptr permissions="prop-default">resB</resptr>
+                    </resptr-prop>
+                    <text-prop name=":hasSimpleText">
+                        <text encoding="utf8">Text</text>
+                        <text encoding="utf8">Text</text>
+                    </text-prop>
+                </resource>
+                <resource label="resC" restype=":TestThing2" id="resD" permissions="res-default">
+                    <resptr-prop name=":hasResource2">
+                        <resptr permissions="prop-default">resB</resptr>
+                    </resptr-prop>
+                </resource>
+        </knora>"""
+    )
+    test_lookup = PropertyTextValueEncodingTypes(
+        formatted_text_props={":hasRichtext"}, unformatted_text_props={":hasSimpleText", ":hasText"}
+    )
+    _check_correctness_all_text_value_encodings(test_ele, test_lookup)
```

Comment:
Here you forgot to check the result. This test will always pass, even with wrong input data!

Reply:
Thanks for spotting it.